toy_2d/src/ilqr_mangonly.py

```diff
@@ -0,0 +1,417 @@
+"""This file tries to solve the unconstrained optimization problem using the iLQR method
+The cost we want to minimize is
+
+           min         sum_{k=0}^{N-1} x_k^T Q x_k + u_k^T R u_k + x_N^T Q_N x_N
+           u_k
+
+        such that   x_{k+1} = f(x,u) + f(x)*lambda
+
+The state vector is 6-dimensional and the control vector is one dimensional in this case. Given an initial guess,
+the problem first performs a rollout and then uses the backward pass to calculate the new controls by linearizing
+the dynamics and solving the riccati equation. 
+
+During the backward pass, in order to capture the effect of lamda in the linearized dynamics we have,
+
+        x_{k+1} = A x_k + B u_k + C lambda_k + d
+        y_k = G x_k + H u_k + J lambda_k + l
+        
+        Also, 0=< lamda_k has a complementarity constraint with y_k=>0
+
+Using this, we first zero out the columns in C and J where lamda_k is zero, then if lamda_k>0 then y_k =0 as
+per the constriant. Hence, 
+        0 = G x_k + H u_k + J lambda_k + l
+        lamda_k = -J^{+}(G x_k + H u_k + l)
+        lamda_k = A' x_k + B' u_k + l'
+
+Substituting this back in original dyanmics eqaution gives us A'' and B'' which is used in the backward pass
+
+        x_{k+1} = A'' x_k + B'' u_k + d''
+
+"""
+
+import numpy as np
+import pdb
+import matplotlib.pyplot as plt
+import pickle
+
+from toy_2d.src import vis_utils
+from toy_2d.src import file_utils
+from toy_2d.src.two_dim_polytope import TwoDimensionalPolytopeParams, \
+                                        TwoDimensionalPolytope
+from toy_2d.src.two_dim_system import TwoDimensionalSystemParams, \
+                                      TwoDimensionalSystem,TwoDSystemMagOnly
+
+from toy_2d.src.two_dim_lcs_approximation import TwoDSystemLCSApproximation
+
+
+class iLQR():
```

Comment:
Why are there two different `iLQR` class definitions?  One in `ilqr_2d.py` and another in `ilqr_mangonly.py` (the latter of which, if it survives a restructuring, I recommend renaming `ilqr_mag_only.py`).  These two classes might even be identically written.  I recommend making one file, e.g. `ilqr.py`, which can handle any system type.

Reply:
Sorry about that. They should be a single class. I have resolved this in the latest commit I will push soon.

Reply:
I have made a single ilqr class which can handle both the 1D and the 2D case and takes the lcs object as an input so as to not use external objects. I have used the system which is present by default in the lcs class to do the forward dynamics.